askama_shared/src/generator.rs

```diff
@@ -20,6 +20,121 @@ pub fn derive_template(input: TokenStream) -> TokenStream {
     }
 }
 
+#[derive(Default)]
```

Comment:
I'd like this refactoring split out in a separate commit, as this commit history makes it very hard to review the changes that were done to add the `config` argument.

Also I'd like the `TemplateArgs` definition and `impl` to go below `build_template()`, in order to be consistent with the rest of the files top-down ordering.

Reply:
Sure!